src/infrastructure/logging.py

```
@@ -0,0 +1,153 @@
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            start_time = time.time()
+            try:
+                result = func(*args, **kwargs)
+                duration = (time.time() - start_time) * 1000
+                logger.info(
+                    f"{func.__name__} completed",
+                    extra={"duration_ms": duration}
+                )
+                return result
+            except Exception as e:
```

Ruff (0.12.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)